client/packages/system/src/Immunisation/api/hooks/useVaccineCourse.ts

```diff
@@ -112,6 +143,62 @@ const useCreate = () => {
     });
   };
 
+  // add iterative mutation for vaccine course items
+  // add iterative mutation for schedule
+
+  return useMutation({
+    mutationFn,
+    onSuccess: () => queryClient.invalidateQueries([VACCINE]),
+  });
+};
+
+const useUpdate = (setErrorMessage: Dispatch<SetStateAction<string>>) => {
+  const { api, storeId, queryClient } = useImmunisationGraphQL();
+  const t = useTranslation('coldchain');
+
+  const mutationFn = async (input: DraftVaccineCourse) => {
+    const apiResult = await api.updateVaccineCourse({
+      input: {
+        id: input.id,
+        name: input.name,
+        demographicIndicatorId: input.demographicIndicatorId,
+        coverageRate: input.coverageRate,
+        isActive: input.isActive,
+        wastageRate: input.wastageRate,
+        doses: input.doses,
+        itemIds: input.vaccineCourseItems ?? [],
+        schedules:
+          input.vaccineCourseSchedules?.map(schedule =>
+            vaccineCourseParsers.toScheduleInput(schedule)
+          ) ?? [],
+      },
+      storeId,
+    });
+
+    const result = apiResult.centralServer.vaccineCourse.updateVaccineCourse;
+
+    // add iterative mutation for vaccine course items
+
+    // add iterative mutation for schedule
+
+    if (result?.__typename === 'VaccineCourseNode') {
+      return result;
+    }
+
+    if (result?.__typename === 'UpdateVaccineCourseError') {
+      // if (result.error.__typename === 'UniqueValueViolation') {
```

Comment:
Are comments still required? I think there aren't any structured errors to manage here...